src/rust/bitbox02-rust/src/hww/api/cardano/sign_transaction/cbor.rs

```diff
@@ -155,16 +155,52 @@ pub fn encode_transaction_body<W: Write>(
                     encode_stake_credential(&mut encoder, keypath)?;
                 }
                 certificate::Cert::StakeDelegation(certificate::StakeDelegation {
-                    keypath,
+                    stake_keypath,
                     pool_keyhash,
                 }) => {
                     if pool_keyhash.len() != 28 {
                         return Err(Error::InvalidInput);
                     }
                     encoder.array(3)?.u8(2)?;
-                    encode_stake_credential(&mut encoder, keypath)?;
+                    encode_stake_credential(&mut encoder, stake_keypath)?;
                     encoder.bytes(pool_keyhash)?;
                 }
+                certificate::Cert::VoteDelegation(certificate::VoteDelegation {
+                    stake_keypath,
+                    r#type,
+                    drep_credhash,
+                }) => {
+                    encoder.array(3)?.u8(9)?;
+                    encode_stake_credential(&mut encoder, stake_keypath)?;
+                    if matches!(r#type,0){
+                        encoder.array(2)?.u8(0)?;
+                        match drep_credhash {
+                            Some(hash) => {
+                                encoder.bytes(hash)?;
```

Comment:
should also check the expected length of the hash.

instead of making it `optional ...`, you could remove that keyword and check the length only. clients that omit it default to the empty bytestring.

Reply:
Added checking of length and I think using `optional` makes more sense here and makes it a bit more readable, maybe just because I dont like to rely on defaults from languages/libraries.

I'm thinking of adding a check in the cases where there should not be a `drep_credhash` sent to check it is none or throw an error. Functionality wise it does not change anything(even if it was sent it will not be encoded), but it might be nice to keep the requests clean. What do you think?

Reply:
Sounds good to me :+1: